bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/handler/IpCameraHandler.java

```diff
@@ -0,0 +1,2020 @@
+            }
+        }
+    }
+
+    // sends direct to ctx so can be either snapshots.mjpeg or normal mjpeg stream
+    public void sendMjpegFirstPacket(ChannelHandlerContext ctx) {
+        final String BOUNDARY = "thisMjpegStream";
+        String contentType = "multipart/x-mixed-replace; boundary=" + BOUNDARY;
+        HttpResponse response = new DefaultHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.OK);
+        response.headers().add(HttpHeaderNames.CONTENT_TYPE, contentType);
+        response.headers().set(HttpHeaderNames.CACHE_CONTROL, HttpHeaderValues.NO_CACHE);
+        response.headers().set(HttpHeaderNames.CONNECTION, HttpHeaderValues.CLOSE);
+        response.headers().add("Access-Control-Allow-Origin", "*");
+        response.headers().add("Access-Control-Expose-Headers", "*");
+        ctx.channel().writeAndFlush(response);
+    }
+
+    public void sendMjpegFrame(byte[] jpg, ChannelGroup channelGroup) {
+        final String BOUNDARY = "thisMjpegStream";
+        ByteBuf imageByteBuf = Unpooled.copiedBuffer(jpg);
+        int length = imageByteBuf.readableBytes();
+        String header = "--" + BOUNDARY + "\r\n" + "content-type: image/jpeg" + "\r\n" + "content-length: " + length
+                + "\r\n\r\n";
+        ByteBuf headerBbuf = Unpooled.copiedBuffer(header, 0, header.length(), StandardCharsets.UTF_8);
+        ByteBuf footerBbuf = Unpooled.copiedBuffer("\r\n", 0, 2, StandardCharsets.UTF_8);
+        streamToGroup(headerBbuf, channelGroup, false);
+        streamToGroup(imageByteBuf, channelGroup, false);
+        streamToGroup(footerBbuf, channelGroup, true);
+    }
+
+    public void streamToGroup(Object msg, ChannelGroup channelGroup, boolean flush) {
+        channelGroup.write(msg);
+        if (flush) {
+            channelGroup.flush();
+        }
+    }
+
+    private void storeSnapshots() {
+        int count = 0;
+        // Need to lock as fifoSnapshotBuffer is not thread safe and new snapshots can be incoming.
+        lockCurrentSnapshot.lock();
+        logger.debug("Storing snapshots now to disk for GIF");
+        for (Object incomingJpeg : fifoSnapshotBuffer) {
+            byte[] foo = (byte[]) incomingJpeg;
+            File file = new File(ffmpegOutputFolder + "snapshot" + count + ".jpg");
+            count++;
+            try {
+                OutputStream fos = new FileOutputStream(file);
+                fos.write(foo);
+                fos.close();
+            } catch (FileNotFoundException e) {
+                logger.error("FileNotFoundException {}", e.getMessage());
+            } catch (IOException e) {
+                logger.error("IOException {}", e.getMessage());
+            }
+        }
+        lockCurrentSnapshot.unlock();
+    }
+
+    public void setupFfmpegFormat(String format) {
+        String inOptions = "";
+        if (ffmpegOutputFolder.equals("")) {
+            logger.error("The camera tried to use a ffmpeg feature when the output folder is not set.");
```

Comment:
The error logging level should be reserved for issues catastrophic enough to threaten the operation of openHab itself. An error in a binding shouldn't qualify for this. Please read the [openHab coding guidelines](https://www.openhab.org/docs/developer/guidelines.html#f-logging) for expected use of logging in bindings.

I recommend changing this to a warning instead.

Reply:
FIXED: Thank you for suggesting, I have gone through all files and made the change to warn.